src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -554,11 +675,60 @@ internal static unsafe uint IUnknown_Release(IntPtr pThis)
             return refcount;
         }
 
+        [UnmanagedCallersOnly]
+        internal static unsafe int IReferenceTrackerTarget_QueryInterface(IntPtr pThis, Guid* guid, IntPtr* ppObject)
+        {
+            ManagedObjectWrapper* wrapper = ComInterfaceDispatch.ToManagedObjectWrapper((ComInterfaceDispatch*)pThis);
+            return wrapper->QueryInterface(in *guid, out *ppObject);
+        }
+
+        [UnmanagedCallersOnly]
+        internal static unsafe uint IReferenceTrackerTarget_AddRefFromReferenceTracker(IntPtr pThis)
+        {
+            ManagedObjectWrapper* wrapper = ComInterfaceDispatch.ToManagedObjectWrapper((ComInterfaceDispatch*)pThis);
+            return wrapper->AddRefFromReferenceTracker();
+        }
+
+        [UnmanagedCallersOnly]
+        internal static unsafe uint IReferenceTrackerTarget_ReleaseFromReferenceTracker(IntPtr pThis)
+        {
+            ManagedObjectWrapper* wrapper = ComInterfaceDispatch.ToManagedObjectWrapper((ComInterfaceDispatch*)pThis);
+            return wrapper->ReleaseFromReferenceTracker();
+        }
+
+        [UnmanagedCallersOnly]
+        internal static unsafe uint IReferenceTrackerTarget_Peg(IntPtr pThis)
+        {
+            ManagedObjectWrapper* wrapper = ComInterfaceDispatch.ToManagedObjectWrapper((ComInterfaceDispatch*)pThis);
+            return wrapper->Peg();
+        }
+
+        [UnmanagedCallersOnly]
+        internal static unsafe uint IReferenceTrackerTarget_Unpeg(IntPtr pThis)
+        {
+            ManagedObjectWrapper* wrapper = ComInterfaceDispatch.ToManagedObjectWrapper((ComInterfaceDispatch*)pThis);
+            return wrapper->Unpeg();
+        }
+
         private static unsafe IntPtr CreateDefaultIUnknownVftbl()
         {
             IntPtr* vftbl = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(ComWrappers), 3 * sizeof(IntPtr));
             GetIUnknownImpl(out vftbl[0], out vftbl[1], out vftbl[2]);
             return (IntPtr)vftbl;
         }
+
+        private static unsafe IntPtr CreateDefaultIReferenceTrackerTargetVftbl()
+        {
+            IntPtr* vftbl = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(ComWrappers), 7 * sizeof(IntPtr));
+            GetIUnknownImpl(out vftbl[0], out vftbl[1], out vftbl[2]);
```

Comment:
The pointers returned by `GetIUnknownImpl` are immediately overridden by next 3 lines. Redundant?

Reply:
Yes. Removed.